File: rust/e1.rs
/*
Find the sum of all the multiples of 3 or 5 below 1000. (fizzbuzz++)
*/

fn euler_1(max_num : usize) -> usize {
    const FIZZERS: [usize; 2] = [3, 5];
    let fizzable = |x| FIZZERS.iter()
                              .any(|n| x % n == 0);
    (0..max_num)
        .filter(|&i| fizzable(i))
        .sum()
}

fn main() {
    println!("{}", euler_1(1000));
}

#[test]
fn test_euler_1() {
    assert_eq!(euler_1(1000), 233168);
}

File: c/e1.c
/*
Find the sum of all the multiples of 3 or 5 below 1000. (fizzbuzz++)
*/
#include <stdio.h>
int MAX_NUM = 1000;

int euler_1(int MAX_NUM) {
    int total = 0;
    for (int i=0; i<MAX_NUM; i++) {
        if (!(i % 3) || !(i % 5)) {
            total += i;
        }
    }
    return total;
}

int main() {
    printf ("%d\n", euler_1(1000));
    return 0;
}
